src/util/markdownToText.ts

```diff
@@ -0,0 +1,240 @@
+        const siblings = tokens.slice(
+          prevHeadingActualIndex + 1,
+          nextHeadingActualIndex === -1 ? undefined : nextHeadingActualIndex,
+        )
+
+        if (token.type === 'list') {
+          // If the list is ordered, we always need a scope if there are any other siblings or if we're at the top level.
+          if (token.ordered) return siblings.length > 1 || (parentDepth === 0 && prevHeadingActualIndex === -1)
+
+          // If the list is unordered, check if there are paragraph siblings
+          return siblings.some(t => t.type === 'paragraph')
+        } else if (token.type === 'table') {
+          // If the table is the only sibling or at the top level, we don't need a scope. Otherwise, we do.
+          return siblings.length > 1 || (parentDepth === 0 && prevHeadingActualIndex === -1)
+        }
+
+        return false
+      }
+
+      switch (token.type) {
+        case 'space':
+          // Ignore
+          break
+        case 'heading':
+          while (stack.length && stack[stack.length - 1] >= token.depth) {
+            stack.pop()
+          }
+          result += `${indent(stack.length)}- ${processInlineTokens(token.tokens ?? token.text)}\n`
+          stack.push(token.depth)
+          break
+        case 'text':
+          result += `${indent(stack.length)}- ${processInlineTokens(
+            'tokens' in token && token.tokens?.length ? token.tokens : token.text,
+          )}\n`
+          break
+        case 'paragraph':
+          // Collapse image paragraphs. This supports multiple images within one paragraph, usually
+          // separated by newline text blocks.
+          if (
+            // First block must be an image
+            token.tokens?.[0].type === 'image' &&
+            // All other blocks must be images or newlines
+            token.tokens.every(token => token.type === 'image' || (token.type === 'text' && token.text === '\n'))
+          ) {
+            processTokens(
+              token.tokens.filter(token => token.type === 'image'),
+              parentDepth + stack.length,
+            )
+            break
+          }
+
+          result += `${indent(stack.length)}- ${processInlineTokens(
+            token.tokens
+              // Replace single newlines with spaces
+              ?.map(token => (token.type === 'text' && token.text === '\n' ? { ...token, text: ' ' } : token)) ??
+              token.text,
+          )}\n`
+          break
+        case 'hr':
+          result += `${indent(stack.length)}- ---\n`
+          break
+        case 'list': {
+          const scope = shouldScope()
+          const currentIndent = scope ? stack.length + 1 : stack.length
+
+          if (scope) {
+            result += `${indent(stack.length)}- ::\n`
+            result += `${indent(stack.length + 1)}- =scope\n`
+          }
+
+          if (token.ordered) {
+            result += `${indent(currentIndent)}- =numbered\n`
```

Comment:
Reflecting on my original suggestion of `=numbered`... 

Do you think it's better to call this `=ordered` for consistency with HTML?

Reply:
Agree, changing to `=ordered` instead.